front/app/containers/Admin/reporting/components/ReportBuilder/Templates/getTemplateData.ts

```diff
@@ -0,0 +1,52 @@
+import { IProjectData } from 'services/projects';
+import { IPhaseData } from 'services/phases';
+
+interface TemplateData {
+  participationMethod: 'ideation' | 'native_survey' | 'other';
+  phaseId?: string;
+}
+
+export const getTemplateData = (
+  project: IProjectData,
+  phases: IPhaseData[]
+): TemplateData => {
+  const hasPhases = project.attributes.process_type === 'timeline';
+
+  if (hasPhases) {
+    for (const phase of phases) {
+      const participationMethod = phase.attributes.participation_method;
+
+      if (
+        participationMethod === 'ideation' ||
+        participationMethod === 'native_survey'
+      ) {
+        return {
+          participationMethod,
+          phaseId: phase.id,
+        };
+      }
+    }
+
+    return {
```

Comment:
Is this return required? Will it just fall back to the same one at the end of the hook? Or does line 36 error if there are no phases?

Reply:
I guess it's not strictly required, if you leave it out it should indeed get to the final return statement. In my mind this is a bit clearer though. If this is a timeline project you already know enough at this point to return, we don't have to go and check the participation method because those are not supposed to be defined for timeline projects- that's only for continuous projects